src/policy/interface.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2014 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_POLICY_INTERFACE_H
+#define BITCOIN_POLICY_INTERFACE_H
+
+#include <string>
+
+class CCoinsViewCache;
+class CScript;
+class CTransaction;
+
+/**
+ * \class CPolicy
+ * Interface class for non-consensus-critical policy logic, like whether or not
+ * a transaction should be relayed and/or included in blocks created.
+ */
+class CPolicy
+{
+public:
+    virtual ~CPolicy() {};
+    virtual bool ApproveScript(const CScript& scriptPubKey) const { return true; };
+    /**
+     * Check for standard transaction types
+     * @return True if all outputs (scriptPubKeys) use only standard transaction forms
+     */
+    virtual bool ApproveTx(const CTransaction& tx, std::string& reason) const { return true; };
```

Comment:
Then I prefer the return true policy simply because it is more convenient when you create custom policies very different from CStandardPolicy (you save the effort to reimplement any method for which you were going to return true, for example, we could have a policy that requires a minimumfee but doesn't have any special requirement for scripts).
As said I can always go back to the pure abstract class, but I don't see what the problem with a non-abstract interface base class (we have something similar in many places, for example BaseSignatureChecker which always returns false).